core/templates/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -264,24 +278,31 @@ angular.module('oppia').directive('questionsList', [
             ctrl.newQuestionSkillIds.forEach(function(skillId) {
               ctrl.linkedSkillsWithDifficulty.push(
                 SkillDifficultyObjectFactory.create(
-                  skillId, '', DEFAULT_SKILL_DIFFICULTY));
+                  skillId, '', null));
             });
             ctrl.showDifficultyChoices = true;
             ctrl.editorIsOpen = true;
             ctrl.initiateQuestionCreation();
           };
+
           ctrl.changeLinkedSkillDifficulty = function() {
-            ctrl.linkedSkillsWithDifficulty.forEach(
-              (linkedSkillWithDifficulty) => {
-                if (!ctrl.newQuestionSkillIds.includes(
-                  linkedSkillWithDifficulty.getId())) {
-                  ctrl.newQuestionSkillIds.push(
-                    linkedSkillWithDifficulty.getId());
-                  ctrl.newQuestionSkillDifficulties.push(
-                    linkedSkillWithDifficulty.getDifficulty());
-                }
-              });
+            if (ctrl.newQuestionSkillIds.length === 1) {
+              ctrl.newQuestionSkillDifficulties = (
+                [ctrl.linkedSkillsWithDifficulty[0].getDifficulty()]);
```

Comment:
Just to check, will `ctrl.linkedSkillsWithDifficulty` always have a 0th element? Can it be empty/undefined/null? If so, do we need a guard clause?

Reply:
No It'll always have one element becuase the question editor can only be open when a skill is selected.